### PR TITLE
completing activity 4

### PR DESCRIPTION
**Pull request by addoadil**
Finishing activity 4 and request review

### CODE REVIEW

File: models/articles.models.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const db = require('../db/connection');
+
+exports.selectArticleById = (article_id) => {
+    return db.query(`SELECT * FROM articles WHERE article_id = $1;`, [article_id]).then((rows) => {
+        if (!rows.rows.length) {
+            return Promise.reject({ status: 404, msg : 'Not found'})
+        }
+        return rows.rows[0]
+    })
```
Comment (johnhutcheon) [inline models/articles.models.js:9]: Nice work!

File: __tests__/endpoints.test.js [hunk 2]
```diff
@@ -51,8 +51,37 @@ describe('GET /api', () => {
...
+describe('GET /api/articles/:article_id', () => {
+  test("Should respond with 404 Not found for an  article_id that does not exist", () => {
+    return request(app)
+      .get('/api/articles/76')
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe('Not found')
+      });
+      });
+});
```
Comment (johnhutcheon) [inline __tests__/endpoints.test.js:83]: Could we add one more test here for an invalid ID. I.e. if we request /api/articles/bananas ?

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hey Adil, nice work! Could you just add the extra test as per my comment please, then resubmit? Thanks! 😄

**Review by roisinoreilly — APPROVED** [review]
great work! merge when ready :)

**Pull request merged**